Framework/Algorithms/src/CreateMonteCarloWorkspace.cpp

```diff
@@ -0,0 +1,163 @@
+// Mantid Repository : https://github.com/mantidproject/mantid
+//
+// Copyright &copy; 2025 ISIS Rutherford Appleton Laboratory UKRI,
+//   NScD Oak Ridge National Laboratory, European Spallation Source,
+//   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
+// SPDX - License - Identifier: GPL - 3.0 +
+#include <algorithm>
+#include <cmath>
+#include <iostream>
+#include <numbers>
+#include <numeric> // For std::accumulate
+#include <random>
+#include <vector>
+
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/Progress.h"
+#include "MantidAPI/Workspace.h"
+#include "MantidAPI/WorkspaceFactory.h"
+#include "MantidAlgorithms/CreateMonteCarloWorkspace.h"
+
+#include "MantidKernel/BoundedValidator.h"
+#include "MantidKernel/Logger.h"
+
+namespace {
+Mantid::Kernel::Logger g_log("CreateMonteCarloWorkspace");
+}
+namespace Mantid {
+namespace Algorithms {
+using Mantid::Kernel::Direction;
+using namespace Mantid::API;
+using namespace std;
+
+// Register the algorithm into the AlgorithmFactory
+DECLARE_ALGORITHM(CreateMonteCarloWorkspace)
+
+//----------------------------------------------------------------------------------------------
+
+/// Algorithms name for identification. @see Algorithm::name
+const std::string CreateMonteCarloWorkspace::name() const { return "CreateMonteCarloWorkspace"; }
+
+/// Algorithm's version for identification. @see Algorithm::version
+int CreateMonteCarloWorkspace::version() const { return 1; }
+
+/// Algorithm's category for identification. @see Algorithm::category
+const std::string CreateMonteCarloWorkspace::category() const { return "Simulation"; }
+
+/// Algorithm's summary for use in the GUI and help. @see Algorithm::summary
+const std::string CreateMonteCarloWorkspace::summary() const {
+  return "Creates a randomly simulated workspace by sampling from the probability distribution of input data.";
+}
+
+//----------------------------------------------------------------------------------------------
+/** Initialize the algorithm's properties.
+ */
+void CreateMonteCarloWorkspace::init() {
+  auto mustBePositive = std::make_shared<Mantid::Kernel::BoundedValidator<int>>();
+  mustBePositive->setLower(0);
+
+  declareProperty(std::make_unique<WorkspaceProperty<MatrixWorkspace>>("InputWorkspace", "", Direction::Input),
+                  "Input Workspace containing data to be fitted");
+  declareProperty("Seed", 32, mustBePositive,
+                  "Integer that initializes a random-number generator, good for reproducibility");
```

Comment:
```suggestion
                  "Integer that seeds initialises the random-number generator, for reproducibility");
```
Mentioning the word "seed" I think gives a lot of clarification.